ethereum/eth2.0-specs
# Add subnet validations for DoS resistance

Addresses #1595 

Quick details for clarity on what was added/changed:
* `beacon_block`
    * prevent multiple beacon blocks from same proposer in a slot
* `committee_index_beacon_attestation`
    * prevent multiple attestations from same participant in a slot
* `aggregate_and_proof`
    * aggregator signs `aggregate_and_proof` and broadcasts `SignedAggregateAndProof`. verify this signature in addition to the proof and attestation sig
    * prevent multiple aggregate_and_proofs from same aggregator in a slot
    * add `DOMAIN_SELECTION_PROOF` and `DOMAIN_AGGREGATE_AND_PROOF`
    * add explicit instructions in validator guide for signing these messages
* reorder gossip conditions to put signature verifications after the cheap checks
